harper-core/src/linting/ive_perfect_progressive.rs

```diff
@@ -0,0 +1,170 @@
+use crate::expr::Expr;
+use crate::expr::SequenceExpr;
+use crate::{Token, TokenKind};
+
+use super::{ExprLinter, Lint, LintKind, Suggestion};
+
+pub struct IvePerfectProgressive {
```

Comment:
While the code is fine the name if the rule is a bit misleading.

It now supports:
- other pronouns than `I`
- other verb contraction than `'ve`

Reply:
I was thinking it doesn't really matter in the code but I think you're right because it's also going to be the name of an option.

Reply:
Yes, that's my concern